crates/typst-cli/src/compile.rs

```diff
@@ -98,12 +98,46 @@ pub fn compile_once(
     }
 
     // Check if main file can be read and opened.
-    if let Err(errors) = world.source(world.main()).at(Span::detached()) {
+    let main_result = world.source(world.main());
+    if main_result.is_err() {
         set_failed();
         if watching {
             Status::Error.print(command).unwrap();
         }
 
+        let errors = match main_result {
+            Err(FileError::InvalidUtf8) => match &command.common.input {
+                Input::Path(path) => {
+                    let extension = path.extension();
+                    let mut errors = main_result.at(Span::detached()).unwrap_err();
+                    if let Some(extension) = extension {
+                        if extension != "typ" {
```

Comment:
```suggestion
                        if extension != "typ" && extension != "typst" {
```
Some people may use `.typst` instead of `.typ`.

Reply:
`.typc` is also a valid typst extension, using for typst code mode.

Reply:
I don't think it is used as a file extension. Moreover, a file with the `typc` extension would not be interpreted as code by the compiler (it is treated just like any regular Typst file: as markup). So I don't think if would make sense to add `typc` as an exception as well.

Reply:
Agreed, typc ist just a highlighting tag.

Reply:
Thanks for your correction.

Reply:
> Some people may use `.typst` instead of `.typ`.

I've only seen that in one specific occasion so far, but is this something we want to officially support?